make_predictions.py

```diff
@@ -130,20 +135,41 @@ def main(_):
   inchikey_op = features[fmap_constants.SPECTRUM_PREDICTION][
       fmap_constants.INCHIKEY]
   ops_to_fetch = [inchikey_op, pred_op]
+  if FLAGS.make_and_save_spectra_plots:
+      true_spectra_op = features[fmap_constants.SPECTRUM_PREDICTION][fmap_constants.DENSE_MASS_SPEC]
+      ops_to_fetch.append(true_spectra_op)
 
   results = {}
+  results_dir = os.path.dirname(FLAGS.output_file)
+  tf.gfile.MakeDirs(results_dir)
 
   def process_fetched_values_fn(fetched_values):
-    keys, predictions = fetched_values
-    for key, prediction in zip(keys, predictions):
-      # Dereference the singleton np string array to get the actual string.
-      key = key[0]
-      results[key] = prediction
+    if not FLAGS.make_and_save_spectra_plots:
+        keys, predictions = fetched_values
+        for key, prediction in zip(keys, predictions):
+            # Dereference the singleton np string array to get the actual string.
+            key = key[0]
+            results[key] = prediction
+    else:
+        keys, predictions, true_spectra = fetched_values
+        for key, prediction, true_spectrum in zip(keys, predictions, true_spectra):
+            # Dereference the singleton np string array to get the actual string.
+            key = key[0]
+            results[key] = prediction
+            spectra_plot_file_name = plot_spectra_utils.name_plot_file(
```

Comment:
can you pull lines 159-168 into a helper function?

Reply:
Done.